### ISSUE TITLE
Error during WebSocket handshake: net::ERR_RESPONSE_HEADERS_TRUNCATED

### ISSUE THREAD

**Issue by luccaheberle** [ratchetphp/Ratchet#658]
I'm having an ocasionally  issue when trying to connect to websocket I get an error message "Error during WebSocket handshake: net::ERR_RESPONSE_HEADERS_TRUNCATED". It happens when I forget to close the connection before closing/refreshing the page, the impression I have is the connectiong stays "open" for that specific ID and when another connection attemp to be stablished it throw this error and close the previous connection.

The new connection receive this header instead of the default: "ratchet detected an invalid reserve code".

Any ideas on this?

**Comment by arvanus** [comment]
The same happens to me.
Relates with https://github.com/ratchetphp/Ratchet/issues/645
Apparently is the same problem.
Any idea?
Thanks!

**Comment by cboden** [comment]
What client are you attempting to connect with?

**Comment by luccaheberle** [comment]
I tried using Chrome and Mozilla firefox to connect, both had the same issue

**Comment by luccaheberle** [comment]
I've made a sample of the server code I'm using which is causing this issue.

```
<?php
namespace MyApp;
use Ratchet\MessageComponentInterface;
use Ratchet\ConnectionInterface;

class chatBot implements MessageComponentInterface {
    protected $clients;

    public function __construct() {
        $this->clients = new \SplObjectStorage;
   }

    public function onOpen(ConnectionInterface $conn) {
        // Store the new connection to send messages to later
        $this->clients->attach($conn);
    }

    public function onMessage(ConnectionInterface $from, $msg) {
		$numRecv = count($this->clients) - 1;
        $from->send($msg);
    }

    public function onClose(ConnectionInterface $conn) {
        // The connection is closed, remove it, as we can no longer send it messages
        $this->clients->detach($conn);

        echo "Connection {$conn->resourceId} has disconnected\n";
    }

    public function onError(ConnectionInterface $conn, \Exception $e) {
        echo "An error has occurred: {$e->getMessage()}\n";

        $conn->close();
    }
}
```

Apache configuration: 
```
    ProxyPass "/ws" "ws://localhost:8080/"
    ProxyPassReverse "/ws" "ws://localhost:8080/"
```

Apache log:

> 
> [Mon Aug 06 09:48:46.210186 2018] [authz_core:debug] [pid 20442] mod_authz_core.c(835): [client 192.168.1.44:53469] AH01628: authorization result: granted (no directives)
> [Mon Aug 06 09:48:46.210280 2018] [proxy:debug] [pid 20442] mod_proxy.c(1123): [client 192.168.1.44:53469] AH01143: Running scheme ws handler (attempt 0)
> [Mon Aug 06 09:48:46.210297 2018] [proxy_ajp:debug] [pid 20442] mod_proxy_ajp.c(722): [client 192.168.1.44:53469] AH00894: declining URL ws://localhost:8080//
> [Mon Aug 06 09:48:46.210309 2018] [proxy_fcgi:debug] [pid 20442] mod_proxy_fcgi.c(972): [client 192.168.1.44:53469] AH01076: url: ws://localhost:8080// proxyname: (null) proxyport: 0
> [Mon Aug 06 09:48:46.210320 2018] [proxy_fcgi:debug] [pid 20442] mod_proxy_fcgi.c(975): [client 192.168.1.44:53469] AH01077: declining URL ws://localhost:8080//
> [Mon Aug 06 09:48:46.210332 2018] [proxy_http:debug] [pid 20442] mod_proxy_http.c(1930): [client 192.168.1.44:53469] AH01113: HTTP: declining URL ws://localhost:8080//
> [Mon Aug 06 09:48:46.210343 2018] [proxy_scgi:debug] [pid 20442] mod_proxy_scgi.c(517): [client 192.168.1.44:53469] AH00865: declining URL ws://localhost:8080//
> [Mon Aug 06 09:48:46.210353 2018] [proxy_wstunnel:debug] [pid 20442] mod_proxy_wstunnel.c(336): [client 192.168.1.44:53469] AH02451: serving URL ws://localhost:8080//
> [Mon Aug 06 09:48:46.210364 2018] [proxy:debug] [pid 20442] proxy_util.c(2203): AH00942: WS: has acquired connection for (localhost)
> [Mon Aug 06 09:48:46.210377 2018] [proxy:debug] [pid 20442] proxy_util.c(2256): [client 192.168.1.44:53469] AH00944: connecting ws://localhost:8080// to localhost:8080
> [Mon Aug 06 09:48:46.210390 2018] [proxy:debug] [pid 20442] proxy_util.c(2426): [client 192.168.1.44:53469] AH00947: connected // to localhost:8080
> [Mon Aug 06 09:48:46.210564 2018] [proxy_wstunnel:debug] [pid 20442] mod_proxy_wstunnel.c(257): [client 192.168.1.44:53469] AH02445: woke from poll(), i=1
> [Mon Aug 06 09:48:46.210591 2018] [proxy_wstunnel:debug] [pid 20442] mod_proxy_wstunnel.c(266): [client 192.168.1.44:53469] AH02446: sock was readable
> [Mon Aug 06 09:48:46.210658 2018] [proxy:debug] [pid 20442] proxy_util.c(2218): AH00943: WS: has released connection for (localhost)
> [Mon Aug 06 09:50:57.203704 2018] [core:debug] [pid 20448] protocol.c(1359): [client 46.243.189.16:44792] AH00569: client sent HTTP/1.1 request without hostname (see RFC2616 section 14.23): login.cgi
> 

Apache version:

> Server version: Apache/2.4.6 (CentOS)

**Comment by hack4mer** [comment]
I solved it and  have posted a solution on #645
